spacy/ml/models/entity_linker.py

```diff
@@ -117,34 +117,20 @@ def empty_kb_factory(vocab: Vocab):
 
 
 @registry.misc("spacy.CandidateGenerator.v1")
-def create_candidates() -> Callable[[KnowledgeBase, Span], Iterable[Candidate]]:
-    return get_candidates
-
-
-@registry.misc("spacy.CandidateBatchGenerator.v1")
-def create_candidates_batch() -> Callable[
-    [KnowledgeBase, SpanGroup], Iterable[Iterable[Candidate]]
+def create_get_candidates() -> Callable[
+    [KnowledgeBase, Iterator[SpanGroup]],
+    Iterator[Iterable[Iterable[Candidate]]],
```

Comment:
Perhaps we can use type aliases across this file as well?

Reply:
Done in https://github.com/explosion/spaCy/pull/12367/commits/8a2a7f1879856af5917182b7b1477568d7117f74.